apps/api/src/persistence/database/arango-query-runner.ts

```diff
@@ -0,0 +1,35 @@
+import { HasId } from '@coscrad/api-interfaces';
+import { isDeepStrictEqual } from 'util';
+import { ICoscradQueryRunner } from '../migrations/coscrad-query-runner.interface';
+import { ArangoDatabase } from './arango-database';
+import { ArangoCollectionId } from './collection-references/ArangoCollectionId';
+import { ArangoDatabaseProvider } from './database.provider';
+import { DatabaseDocument } from './utilities/mapEntityDTOToDatabaseDTO';
+
+export class ArangoQueryRunner implements ICoscradQueryRunner {
+    private readonly arangoDatabase: ArangoDatabase;
+
+    constructor(arangoDatabaseProvider: ArangoDatabaseProvider) {
+        this.arangoDatabase = arangoDatabaseProvider.getDBInstance();
+    }
+
+    async update<TOldDocument extends DatabaseDocument<HasId>, UNewDocument>(
```

Comment:
Cool typings

Reply:
But the `UNewDocument` should also have `_key` on it or else Arango will miss the query.